Inline Math.clamp Polyfill

JustinFreitas opened the pull request:
Unsure where Math.clamped was defined, so instead, let's inline an implementation.

sneat commented:
Realistically, I think we can just remove this polyfill entirely, Math.clamp has been supported for a long time https://caniuse.com/css-math-functions

JustinFreitas commented:
Different clamp, I think. MDN MathMath - JavaScript | MDNdeveloper.mozilla.orgOn Aug 3, 2025, at 7:06 PM, Blair ***@***.***> wrote:﻿sneat left a comment (NecroticGnome/ose-foundry-core#99)
Realistically, I think we can just remove this polyfill entirely, Math.clamp has been supported for a long time https://caniuse.com/css-math-functions

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

sneat commented:
You're spot on (whoops).

I did a little more digging though and it's part of Foundry:
<img width="629" height="323" alt="image" src="https://github.com/user-attachments/assets/4729ec29-c55a-403a-92d8-acad1ab71655" />

The code was added to deal with Math.clamp being new, but Math.clamped being old. Do you mind please adding the following to the polyfill so we can detect it being used in the future?

```js
foundry.utils.logCompatibilityWarning("Math.clamp is using a polyfill method provided by the OSE system.", {once: true});
```

JustinFreitas commented:
Ah, yes... I agree.  I didn't even notice the clamp definition... I was tunnel visioning on the clamped function.  I'll get rid of the polyfill and update the pull request.  Thanks for pointing that out.

sneat commented:
Thank you @JustinFreitas - I appreciate the conversation and your efforts.

JustinFreitas commented:
It's fun to chip in.  Ok, I deleted the file (it was the only function in the file so not necessary anymore) and removed the import and call of it in ose.js.  This should do the trick.  If you notice anything else, just say and I'll make the change.  Otherwise, happy merging!
